lib/Provider/AtLoginProvider.php

```diff
@@ -0,0 +1,50 @@
+<?php
+declare(strict_types = 1);
+
+namespace OCA\TwoFactorEmail\Provider;
+
+use OCP\Authentication\TwoFactorAuth\ILoginSetupProvider;
+use OCP\Template;
+use OCP\IUser;
+use OCA\TwoFactorEmail\Service\Email as EmailService;
+use OCP\IConfig as OCCONFIG;
+use OCP\Authentication\TwoFactorAuth\IRegistry;
+
+class AtLoginProvider implements ILoginSetupProvider {
+
+	public function __construct(IUser $myUser, $mySecret, EmailService $EmailService, OCCONFIG $occonfig, IRegistry $registry, Email $provider) {
+		$this->myUser = $myUser;
+		$this->mySecret = $mySecret;
+		$this->EmailService = $EmailService;
+		$this->occonfig = $occonfig;
+		$this->registry = $registry;
+		$this->provider = $provider;
+	}
+
+	public function setEnabledActivity() {
+        // With OCP\IConfig i can get all configured value in config/config.php so i can know if twofactor is forced
+        // And it's not in database
+		$state2fa = $this->occonfig->getSystemValue('twofactor_enforced');
+
+        //If true, i set plugin " enabled " in databse for the user
+        //In database is : oc_twofactor_providers , and set enabled to 1 , 0 is disable
+        //For futur, if needed, disableProviderFor exist :)
+		if ($state2fa) {
+			$this->registry->enableProviderFor($this->provider, $this->myUser);
+		}
+	}
+
+	public function getBody(): Template {
+        //The code is from Email.php, getTemplate
```

Comment:
You copied the Provider/Email.php logic (=authentication). This is wrong since you must first VERIFY that the code is correct. Currently, any number entered is accepted to activate twofactor_email. Thus, you should implement Service/SetupService.php.